making_change/making_change.py

```diff
@@ -3,7 +3,100 @@
 import sys
 
 def making_change(amount, denominations):
-  pass 
+  cache = [0 for i in range(amount + 1)]
+  cache[0] = 1
+
+  #loop over the denominations list
+  for coin in denominations:
+    for higher_amount in range(coin, amount + 1):
+      remainder = higher_amount - coin
+      cache[higher_amount] += cache[remainder]
+
+  return cache[amount]
+
+  print(making_change(20, denominations)) 
+
+
+
+# denominations = {
+#   'penny': 1,
+#   'nickel': 5,
+#   'dime': 10,
+#   'quarter': 25,
+#   'half_dollar': 50
+# }
+
+# def making_change(amount, denominations):
```

Comment:
Good idea!
It's always nice to leave your old work so you can see your own progress.